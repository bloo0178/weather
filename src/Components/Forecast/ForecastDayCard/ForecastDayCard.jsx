import React from "react";
import styles from "./ForecastDayCard.module.css";

class ForecastDayCard extends React.Component {

  render() {
    const { date, tempUnit, times } = this.props;
    return (
      <div className={styles.Container}>
        <h4 className={styles.ForecastDate}>{date}</h4>
        <table>
            <tbody className={styles.ForecastDetail}>
          {times.map(time => {
            let regex = /(.{3})(?=(\.png))/;
            let icon = regex.exec(time.icon)[0];
            const iconSrc = require(`../../../common/icons/${icon}.svg`);
            return (
              <tr key={time.time}>
                <td>{time.time}</td>
                <td>
                  <img src={iconSrc} alt={`${time.time} weather icon`} />
                </td>
                <td>{time.temp + " " + tempUnit}</td>
                {/*<td>{time.description}</td>*/}
              </tr>
            );
          })}
          </tbody>
        </table>
      </div>
    );
  }
}

export default ForecastDayCard;
