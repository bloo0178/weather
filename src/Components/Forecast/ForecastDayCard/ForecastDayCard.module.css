.Container {
  width: 90%;
  border: 3px solid black;
  border-radius: 10px;
  background-color: black;
  opacity: 1;
}

.ForecastDate {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.ForecastDetail {
  text-align: center;
}

table {
  width: 100%;
}

h4 {
  color: white;
}

tr:nth-child(odd) {
  background-color: #1f1f1f;
  color:white;
}

tr:nth-child(even) {
  background-color: black;
  color: white;
}
