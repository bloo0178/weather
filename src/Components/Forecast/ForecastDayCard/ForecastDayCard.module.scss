.forecastWrapper {
  width: 75vw;
  padding-top: 1rem;
}

.forecastDate {
  padding-bottom: 1rem;
}

.forecastDetail {
  text-align: center;
}
