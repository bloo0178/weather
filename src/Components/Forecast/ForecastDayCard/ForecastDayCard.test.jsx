import React from "react";
import ForecastDayCard from "./ForecastDayCard";
import { shallow } from "enzyme";

let times = [
	{
		time: "12am",
		icon: "http://openweathermap.org/img/w/10n.png",
		temp: "85"
	},
	{
		time: "3am",
		icon: "http://openweathermap.org/img/w/10n.png",
		temp: "86"
	},
	{
		time: "6am",
		icon: "http://openweathermap.org/img/w/10n.png",
		temp: "87"
	}
];

describe("<ForecastDayCard />", () => {
	test("renders correctly", () => {
		let wrapper = shallow(
			<ForecastDayCard
				date={"Tue May 14"}
				tempUnit="F"
				times={times}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
