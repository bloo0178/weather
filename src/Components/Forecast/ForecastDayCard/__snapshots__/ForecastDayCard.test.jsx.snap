// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastDayCard /> renders correctly 1`] = `
<div
  className="Container"
>
  <h4
    className="ForecastDate"
  >
    Tue May 14
  </h4>
  <table>
    <tbody
      className="ForecastDetail"
    >
      <tr
        key="12am"
      >
        <td>
          12am
        </td>
        <td>
          <img
            alt="12am weather icon"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "10n.svg",
              }
            }
          />
        </td>
        <td>
          85 F
        </td>
      </tr>
      <tr
        key="3am"
      >
        <td>
          3am
        </td>
        <td>
          <img
            alt="3am weather icon"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "10n.svg",
              }
            }
          />
        </td>
        <td>
          86 F
        </td>
      </tr>
      <tr
        key="6am"
      >
        <td>
          6am
        </td>
        <td>
          <img
            alt="6am weather icon"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "10n.svg",
              }
            }
          />
        </td>
        <td>
          87 F
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
